pulp_ansible/app/galaxy/v3/viewsets.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import viewsets
+
+from drf_spectacular.utils import OpenApiParameter, extend_schema, extend_schema_view
+
+from pulp_ansible.app.galaxy.v3.pagination import LimitOffsetPagination
+
+from pulp_ansible.app.galaxy.v3.serializers import (
+    CollectionVersionSearchListSerializer,
+)
+
+from pulp_ansible.app.models import CrossRepositoryCollectionVersionIndex, AnsibleDistribution
+
+from pulp_ansible.app.galaxy.v3.filters import CollectionVersionSearchFilter
+from pulp_ansible.app.tasks.collectionversion_index import rebuild_index
+
+from pulpcore.plugin.tasking import dispatch
+from pulpcore.plugin.viewsets import OperationPostponedResponse
+
+
+@extend_schema_view(
+    list=extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="repository",
+                type={"type": "array", "items": {"type": "string"}},
+                location=OpenApiParameter.QUERY,
+                required=False,
+                style="form",
+                explode=True,
+                description="Filter collectionversions that are in these repository ids.",
+            ),
+            OpenApiParameter(
+                name="repository_name",
+                type={"type": "array", "items": {"type": "string"}},
+                location=OpenApiParameter.QUERY,
+                required=False,
+                style="form",
+                explode=True,
+                description="Filter collectionversions that are in these repositories.",
+            ),
+            OpenApiParameter(
+                name="distribution",
+                type={"type": "array", "items": {"type": "string"}},
+                location=OpenApiParameter.QUERY,
+                required=False,
+                style="form",
+                explode=True,
+                description="Filter collectionversions that are in these distrubtion ids.",
+            ),
+            OpenApiParameter(
+                name="distribution_base_path",
+                type={"type": "array", "items": {"type": "string"}},
+                location=OpenApiParameter.QUERY,
+                required=False,
+                style="form",
+                explode=True,
+                description="Filter collectionversions that are in these base paths.",
+            ),
+        ],
+    )
+)
+class CollectionVersionSearchViewSet(viewsets.ModelViewSet):
+    """
+    A viewset for cross-repo searches.
+    """
+
+    serializer_class = CollectionVersionSearchListSerializer
+    pagination_class = LimitOffsetPagination
+    filterset_class = CollectionVersionSearchFilter
+
+    # ordering_fields = ("collection_version.pulp_created",)
+
+    def get_queryset(self):
+        return (
+            CrossRepositoryCollectionVersionIndex.objects.select_related("repository")
+            .select_related("collection_version")
+            .select_related("repository_version")
+            .select_related("namespace_metadata")
+            .all()
```

Comment:
This is a massive query. How is the performance?

Reply:
![image](https://user-images.githubusercontent.com/1869705/224759305-362d4164-d9d3-4b20-b377-6ce05ef9c804.png)

Performance is on par with other endpoints in the system as the CV count grows.  Uploads and syncs increase in duration, but I don't have baseline data to determine if that was always the case prior to this PR.